### ISSUE TITLE
Webpack metrics

### ISSUE THREAD

**Issue by g8tguy** [gatsbyjs/gatsby#616]
Is there a way to set a more verbose output during the build.
It would be nice to see some build statistics, like:
https://github.com/FormidableLabs/webpack-dashboard
https://github.com/robertknight/webpack-bundle-size-analyzer

- each included lib size
- final bundle size
- build time
- cpu usage
- etc

And is it possible to use a `hard-source-webpack-plugin` with gastby to speed up the build time a bit?

**Comment by g8tguy** [comment]
As it turns out , **hard-source-webpack-plugin** is actually working pretty well



```shell
npm install node-object-hash hard-source-webpack-plugin --save-dev
```
`gatsby-node.jsx`

```js
import HardSourceWebpackPlugin from 'hard-source-webpack-plugin'

exports.modifyWebpackConfig = (config, stage) => {

    let plugin = new HardSourceWebpackPlugin({
        cacheDirectory: 'cache/[confighash]',
        recordsPath: 'cache/[confighash]/records.json',
        configHash: (webpackConfig) => {
            return require('node-object-hash')().hash(webpackConfig);
        }
    })

    config.merge({
        plugins: [plugin]
    })

```
Still tweaking it, but from the get go it speeds up things pretty well actually on an intell i7 machine with ssd.

**Comment by g8tguy** [comment]
got `webpack-dashboard` working. with this config:

```js
import Dashboard from'webpack-dashboard'
import DashboardPlugin from'webpack-dashboard/plugin'

const dashboard = new Dashboard()

    config.merge({
        quiet: true,
        plugins: [new DashboardPlugin(dashboard.setData)]
    })
```

but it's breaking on build error, because `quiet: true,` for some reason has no effect. Weback keeps spamming the stdout with log messages.

Maybe there is other way to enable the `quiet: true,` modifier ?

**Comment by KyleAMathews** [comment]
I'd love to see Gatsby's output prettied up and made more useful.

I want to like https://github.com/robertknight/webpack-bundle-size-analyzer but it's kind of a bummer it works on pre-minified/prebundled code. Perhaps we could use the guts of https://www.npmjs.com/package/source-map-explorer to get the real sizes and output those.

Re: hard-source-webpack-plugin — I had it enabled on the 1.0 branch for a while but eventually disabled it as I kept running into weird errors. It does seem super promising though — I saw 30-45% speedups pretty consistently — but too early for general distribution. Would love to have more people using it and helping make it better. With the plugin system in 1.0, we could add a plugin for this to make it easy for people to try.
